### DISCUSSION TITLE
Event Lane Types

### DISCUSSION THREAD

**Discussion by schnuerle** [openmobilityfoundation/curb-data-specification#37] [Data Specification]
What is the enumerated list of the lane types that are possible in Events?

https://github.com/openmobilityfoundation/curb-data-specification/blob/feature-release-work-1/events/README.md#lane-type

@lummish will provide the first pass at these.

**Comment by lummish** [comment]
The original draft proposal included the following lane types:

- `traffic_lane`: A standard vehicle traffic lane
- `turn_lane`: A dedicated turn lane
- `bike_lane`: A lane dedicated for usage by cyclists
- `unspecified`: Unspecified

We should probably also add `bus_lane` type. Beyond that, it may also make sense to replace `turn_lane` with two different values: `right_turn_lane` and `left_turn_lane`. I'm not sure whether or not this would be useful for the purpose of analysis.

**Reply by schnuerle** [reply]
I added these here along with bus_lane, and rand with turn_lane for now unless there is a case for right and left (at which point we'd probably still need turn_lane to mean one where left or right is unspecified). There also could be left AND right for when a street dead ends or hits a one way in the opposite direction. Maybe even forward OR right/left. So that detail might not be needed for the complexity. 

https://github.com/openmobilityfoundation/curb-data-specification/blob/feature-release-work-1/events/README.md#lane-type

**Comment by jlarsonOmahaNE** [comment]
I think that traffic_lane should be renamed to travel_lane as we use this as our standard. This could then be matched/tied into the WZDx lane types. See below.
https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/enumerated-types/LaneType.md

I also think that with a tie in, you would then be able to use the WZDx Road Restrictions to help with the curb activity_type (such as the snow plow/removal example brought up today or other utility use cases). 
https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/enumerated-types/RoadRestriction.md

**Reply by schnuerle** [reply]
Would love to tie in with WZDx! 

traffic_lane -> travel_lane seems good. How does that align to the `LaneType` from WZDx though?  They don't use travel_lane.

It does seem like adding their `shoulder` and `sidewalk` values for `lane_type` would be useful.

Looks like they may be updating this list in this branch for 4.0: https://github.com/usdot-jpo-ode/wzdx/blob/v4.0-lane-types/spec-content/enumerated-types/LaneType.md

For Road Restrictions, where do you see snow removal? This list of restrictions seems different than activity types.

**Reply by jlarsonOmahaNE** [reply]
My thought process was to change traffic to travel then use some of their LaneTypes to include into these. Sorry that wasn't clear.

Under road restrictions, snow removal wasn't in their specs but I was thinking that CDS lane_type could somehow tie into the values of Road Restriction. An example for us is that some of our streets have a snow emergency route or street sweeping that affect the curb. I was mainly trying to think how to keep the CDS lane_type and activity_type generalized but connect to/create road restrictions for an event like this.

**Reply by schnuerle** [reply]
I pulled some of the lane types from WZDx 4.0 to the CDS spec, but not 1:1 yet. https://github.com/openmobilityfoundation/curb-data-specification/blob/feature-release-work-1/events/README.md#lane-type

Road restrictions seems useful but maybe not essential for v1.0? I'm going to open an Issue here to talk about WZDx alignment more broadly.

**Reply by cartograham** [reply]
I think it might be good to consider adding a optional lane “sub type” field to allow agencies to populate with more detailed information, if they have it. 

As an example, you’ve indicated that bike lane is one of the lane types, but as you know there are a wide array of bike lane infrastructure (conventional, protected bike lanes, etc). By their nature some of these specific types prohibit parking because the facility is blocking that part of the curb. It would be valuable information to convey if the agency had it.

**Reply by schnuerle** [reply]
Let's put a pin in sub type and see how this could be used and how much it's needed for CDS.

Also I made discussion are specific to WZDx here: https://github.com/openmobilityfoundation/curb-data-specification/discussions/53
